src/sqlparser/src/ast/mod.rs

```diff
@@ -96,7 +97,10 @@ pub struct Ident {
 
 impl Ident {
     /// Create a new identifier with the given value and no quotes.
-    pub fn new<S>(value: S) -> Self
+    /// In situations like construct a identifier from existing object
+    /// names, we may prefer to use this 'cause the input string is
+    /// guanteed to be not empty.
+    pub fn new_safe<S>(value: S) -> Self
```

Comment:
I don't think it's necessary to add `_safe` to this function, which makes this PR harder to review. Adding some doc comments should be enought.

BTW, `_unchecked` might be better than `_safe` (it's actually unsafe..? 😄)

Reply:
I was dumb...